common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -345,6 +346,116 @@ def make_asset_id(course_id, asset_xml):
+
+        create_if_not_present: If True, then a new library is created if it doesn't already exist.
+            Otherwise, it throws an InvalidLocationError if the library does not exist.
+
+        default_class, load_error_modules: are arguments for constructing the XMLModuleStore (see its doc)
+    """
+
+    # We only actually support one library import at a time right now.
+    if len(library_dirs) > 1:
+        raise NotImplementedError(
+            "Only one library may be imported at a time from this function.")
+    lib_dir = path(data_dir) / library_dirs[0]
+    lib_file = open(lib_dir / 'library.xml')
+
+    xml_module_store = XMLModuleStore(
+        data_dir,
+        default_class=default_class,
+        course_dirs=library_dirs,
+        load_error_modules=load_error_modules,
+        xblock_mixins=store.xblock_mixins,
+        xblock_select=store.xblock_select,
+        library=True,
+    )
+
+    try:
+        library = store.get_library(target_library_id)
+    except (ItemNotFoundError, AssertionError):
+        library = None
+    if library is None and create_if_not_present:
+        library = store.create_library(
+            org=target_library_id.org,
+            library=target_library_id.library,
+            user_id=user_id,
+            fields={"display_name": display_name}
+        )
+
+    logger, errors = make_error_tracker()  # pylint: disable=unused-variable
+    system = ImportSystem(
+        xml_module_store, library.location.library_key, lib_dir,
+        logger,
+        load_error_modules=load_error_modules,
+        mixins=xml_module_store.xblock_mixins,
+        field_data=KvsFieldData(kvs=DictKeyValueStore()))
+
+    # pylint: disable=no-member
+    lib_data = etree.parse(lib_file, parser=edx_xml_parser).getroot()
+
+    # pylint: disable=no-member
+    temp_library = system.process_xml(etree.tostring(lib_data, encoding='unicode'))
+    # Return a list to make return signature similar to course import. Also to permit multi import in
+    # the future if it's needed.
+
+    # Everything that has happened so far has only placed the data in a temporary position. We now move it
+    # over to its destination library. First, delete the old items.
+    for child in library.children:
+        store.delete_item(child, user_id)
+
+    temp_key = temp_library.location.library_key
+    library_key = library.location.library_key
+    # Now stick all the new items into the library.
+    for module in temp_library.children:
+        _import_module_and_update_references(
+            xml_module_store.get_item(module), store, user_id,
+            temp_key,
+            library_key,
+            do_import_static=do_import_static,
+            runtime=library.runtime,
+            library=True
+        )
+
+    if do_import_static:
+        import_static_content(
+            lib_dir, static_content_store,
+            library_key, subpath='static', verbose=False)
+
+    return [library]
```

Comment:
You don't have to refactor the whole module. You could just refactor the functions that change depending on whether they're libraries or not, and leave thin wrapper functions to allow the callers to remain unchanged for the moment.